lib/media.js

```diff
@@ -97,10 +97,16 @@ exports.setTrack = function (item, callback) {
       callback(err)
       return
     }
-    var duration = 0
+    var duration;
     if (item.contentDetails && item.contentDetails.duration) {
       duration = util.parsePeriod(item.contentDetails.duration)
     }
+    // return duration in milliseconds
+    duration = duration * 1000;
+    // if we have a negative or 0 duration omit it altogether
+    if (duration < 1) {
```

Comment:
> ...and then we don't need this check

We still need that check regardless, if the util.parsePeriod returns 0, no amount of multiplication will make it better. If for some reason it manages to return a negative duration then it still won't make it better so the check is needed.